### DISCUSSION TITLE
Scalar filtering optimization

### DISCUSSION THREAD

**Discussion by YolandaLyj** [milvus-io/milvus#38756] [Q&A and General discussion] [answered]
I saw this quote in the [blog.](https://zilliz.com/learn/attribute-filtering). I don't quite understand what it means, can you tell me more about it.

"So if we can figure which buckets to compute, we can just filter vectors in those buckets to greatly reduce computation load. "

Is this optimization now implemented？

**Comment by yhmo** [comment, marked as answer]
From v2.4.8, Milvus provides a new feature "clustering key":
https://milvus.io/docs/clustering-compaction.md#Clustering-Compaction

You can define a clustering key in the collection schema:
```
schema = MilvusClient.create_schema()
schema.add_field(field_name="dummy", datatype=DataType.INT32, is_clustering_key=True)
```

With this clustering key, data is arranged by the values of the "dummy" field. When you call search(filter="dummy == 8"), the search engine can quickly know which segment contains the data with "dummy == 8", other segments are skipped, to improve the search performance.

**Reply by YolandaLyj** [reply]
Thank you. That's great!
